app/scripts/metamask-controller.js

```diff
@@ -661,12 +662,21 @@ module.exports = class MetamaskController extends EventEmitter {
     if (this.isNetworkLoading()) this.lookupNetwork()
   }
 
-  setRpcTarget (rpcTarget) {
-    this.configManager.setRpcTarget(rpcTarget)
+  setDefaultRpc () {
+    this.configManager.setRpcTarget('http://localhost:8545')
     extension.runtime.reload()
     this.lookupNetwork()
   }
 
+  setCustomRpc (rpcTarget, rpcList) {
+    this.configManager.setRpcTarget(rpcTarget)
+    return this.preferencesController.updateFrequentRpcList(rpcTarget)
+      .then(() => {
+        extension.runtime.reload()
```

Comment:
Worth noting that *eventually* we'll need to move the `extension` references outside of `MetaMaskController`, so that it is not platform opinionated.